### DISCUSSION TITLE
CC1101 - Pocsag - No TX Output

### DISCUSSION THREAD

**Discussion by mkrebbers** [jgromes/RadioLib#911] [General]
Hey There,

I am working on a Pocsag transmitter with ESP32 and CC1101. However i cant get TX output from CC1101 using the example. 
Transmitting in general is working properly (AX25,FSK,..). To check if there is any transmittion i am using SDR with dedicated antenna but the waterfall keeps flat. 

Pins checked, Radio initialized, CLI shows "transmitted"

Any ideas on that? 

Marc

**Comment by jgromes** [comment]
What does the code look like? Are there any errors? And did you check output in debug mode?

**Reply by mkrebbers** [reply]
This is the Code from the Example just modified to the CC1101. At the moment i cant see a problem in the main.cpp.

Waterfall:
![grafik](https://github.com/jgromes/RadioLib/assets/59209815/4ee33ca3-8862-46ab-a039-7188f53ad3fd)

```c++
/*
   RadioLib Pager (POCSAG) Transmit Example

   This example sends POCSAG messages using SX1278's
   FSK modem.

   Other modules that can be used to send POCSAG:
    - SX127x/RFM9x
    - RF69
    - SX1231
    - CC1101 used
    - SX126x
    - nRF24
    - Si443x/RFM2x
    - SX128x

   For default module settings, see the wiki page
   https://github.com/jgromes/RadioLib/wiki/Default-configuration

   For full API reference, see the GitHub Pages
   https://jgromes.github.io/RadioLib/
*/

// include the library
#include <RadioLib.h>

// CC1101 has the following connections:
// NCS pin:    5
// GDO0 pin:  4
// RST pin:   NC
// GDO2 pin:  2
CC1101 radio = new Module(5, 4, RADIOLIB_NC, 2);


// create Pager client instance using the FSK module
PagerClient pager(&radio);

void setup() {
  Serial.begin(9600);

  // initialize CC1101 with default settings
  Serial.print(F("[CC1101] Initializing ... "));
  int state = radio.begin();

  if(state == RADIOLIB_ERR_NONE) {
    Serial.println(F("success!"));
  } else {
    Serial.print(F("failed, code "));
    Serial.println(state);
    while(true);
  }

  // initialize Pager client
  Serial.print(F("[Pager] Initializing ... "));
  // base (center) frequency:     440.0 MHz testing!
  // speed:                       1200 bps
  state = pager.begin(440.0, 1200);
  if(state == RADIOLIB_ERR_NONE) {
    Serial.println(F("success!"));
  } else {
    Serial.print(F("failed, code "));
    Serial.println(state);
    while(true);
  }
}

void loop() {
  Serial.print(F("[Pager] Transmitting messages ... "));

  // the simples form of "message" is just a tone on the destination pager
  int state = pager.sendTone(1234567);

  if(state == RADIOLIB_ERR_NONE) {
    Serial.println(F("success!"));
  } else {
    Serial.print(F("failed, code "));
    Serial.println(state);
  }

  // wait for a second before transmitting again
  delay(1000);
}
```

This is the Output on Serial: 

 RadioLib Debug Info
 Version:  6.3.0.0
 Platform: ESP32
 Compiled: Dec 21 2023 14:57:11
 M        CC1101
                                                   success!
[Pager] Initializing ... success!
[Pager] Transmitting messages ... success!

**Reply by jgromes** [reply]
Could you enable both debug nad verbose? That will show all the SPI transactions.

**Reply by mkrebbers** [reply]
Sure! I inserted a while so that the sendTone was only processed one time and then paused with the state "RADIOLIEB_ERR_NONE".
With verbose mode i cant see the success message on transmitting. If i switch it off again everything is normal. 
The phrases from CMD     W       35      0F keep on repeating in the console. 

This is what came out: 

[CC1101] Initializing ... 
RadioLib Debug Info
Version:  6.3.0.0
Platform: ESP32
Compiled: Dec 21 2023 17:01:41

R       71      14
M       CC1101
CMD     W       30      2F
CMD     W       36      0F
R       75      1
R       18      4
W       18      14
R       18      14
R       18      14
W       18      14
R       18      14
R       2       3F
W       2       2E
R       2       2E
R       0       29
W       0       2E
R       0       2E
R       7       4
W       7       4
R       7       4
R       8       45
W       8       5
R       8       5
R       8       5
W       8       5
R       8       5
CMD     W       36      0F
R       D       1E
W       D       10
R       D       10
R       E       C4
W       E       B1
R       E       B1
R       F       EC
W       F       3B
R       F       3B
R       3E      C6
W       3E      C0
R       3E      C0
CMD     W       36      0F
R       10      8C
W       10      87
R       10      87
R       11      22
W       11      83
R       11      83
CMD     W       36      0F
R       10      87
W       10      F7
R       10      F7
CMD     W       36      0F
R       15      47
W       15      17
R       15      17
R       15      17
W       15      14
R       15      14
R       3E      C0
W       3E      C0
R       3E      C0
R       8       5
W       8       5
R       8       5
R       6       FF
W       6       3F
R       6       3F
R       7       4
W       7       64
R       7       64
R       13      22
W       13      2
R       13      2
CMD     W       36      0F
R       75      1
R       12      2
W       12      2
R       12      2
CMD     W       36      0F
R       75      1
R       12      2
W       12      2
R       12      2
R       8       5
W       8       5
R       8       5
R       12      2
W       12      2
R       12      2
R       4       D3
W       4       12
R       4       12
R       5       91
W       5       AD
R       5       AD
CMD     W       3A      0F
CMD     W       3B      0F
success!
[Pager] Initializing ... CMD    W       36      0F
R       75      1
R       12      2
W       12      2
R       12      2
R       8       5
W       8       5
R       8       5
CMD     W       36      0F
R       75      1
R       12      2
W       12      2
R       12      2
CMD     W       36      0F
R       15      14
W       15      4
R       15      4
R       15      4
W       15      0
R       15      0
success!
[Pager] Transmitting messages ... W     D       10
W       E       EC
W       F       43
CMD     W       35      0F
CMD     W       36      4F
R       2       2E
W       2       B
R       2       B
R       0       2E
W       0       C
R       0       C
R       8       5
W       8       15
R       8       15
R       8       15
W       8       16
R       8       16
CMD     W       35      0F
W       D       10
W       E       EC
W       F       59
CMD     W       35      2F
CMD     W       36      2F
R       2       B
W       2       B
R       2       B
R       0       C
W       0       C
R       0       C
R       8       16
W       8       16
R       8       16
R       8       16
W       8       16
R       8       16
CMD     W       35      0F

**Reply by jgromes** [reply]
@mkrebbers thank you, this made everything clear. There was a bug that caused direct transmission to not work correctly on CC1101 (instead of exiting immediately after changing the frequency, the library tried to re-eanble direct mode). I pushed a fix for this and testing on my CC1101 setup, it seems to be working correctly, though please let me know if you find any other issues.

**Reply by mkrebbers** [reply]
@jgromes thank you for fixing it so fast! For me its working with C1101 and also the Pager is decoding.
